### PR TITLE
[build]: allow single src file to build multiple independent debian p…

### PR DESCRIPTION
**Pull request by lguohan**
…ackages

add_derived_package setup dependency between the main deb and derived deb.
The dervied deb depends on the main deb and need to install the main deb.

add_peer_package does not setup dependency between the main deb and peer deb,
does not require to install the main deb.

### CODE REVIEW

**Comment by lguohan** [comment]
@marian-pritsak , add_derived_package does not work this case as it will try to install the base deb pacakge. However, I am not sure add_peer_pacakge is the best way to address problem.

**Comment by marian-pritsak** [comment]
@lguohan, can you please describe the problem that you have and give me a usage example, because it is not so clear from PR

File: slave.mk [hunk 1]
```diff
@@ -140,7 +140,7 @@ $(addprefix $(DEBS_PATH)/, $(SONIC_ONLINE_FILES)) : $(DEBS_PATH)/% : .platform
 $(addprefix $(DEBS_PATH)/, $(SONIC_MAKE_DEBS)) : $(DEBS_PATH)/% : .platform $$(addsuffix -install,$$(addprefix $(DEBS_PATH)/,$$($$*_DEPENDS)))
 	$(HEADER)
 	# remove target to force rebuild
-	rm -f $(addprefix $(DEBS_PATH)/, $* $($*_DERIVED_DEBS))
+	rm -f $(addprefix $(DEBS_PATH)/, $* $($*_DERIVED_DEBS) $($*_PEER_DEBS))
```
Comment (marian-pritsak) [inline slave.mk]: Same goes for online debs

File: slave.mk [hunk 2]
```diff
@@ -351,12 +364,18 @@ SONIC_CLEAN_DEBS = $(addsuffix -clean,$(addprefix $(DEBS_PATH)/, \
...
 		   $(SONIC_PYTHON_STDEB_DEBS) \
-		   $(SONIC_DERIVED_DEBS)))
+		   $(SONIC_DERIVED_DEBS) \
+		   $(SONIC_PEER_DEBS)))
 $(SONIC_CLEAN_DEBS) : $(DEBS_PATH)/%-clean : .platform $$(addsuffix -clean,$$(addprefix $(DEBS_PATH)/,$$($$*_DERIVED_FROM)))
 	@# remove derived targets if main one is removed, because we treat them
 	@# as part of one package
 	@rm -f $(addprefix $(DEBS_PATH)/, $* $($*_DERIVED_DEBS))
 
+$(SONIC_CLEAN_DEBS) : $(DEBS_PATH)/%-clean : .platform $$(addsuffix -clean,$$(addprefix $(DEBS_PATH)/,$$($$*_MAIN_DEB)))
```
Comment (marian-pritsak) [inline slave.mk]: No need for separate pattern rule
Just extend rule above

Comment (lguohan) [inline slave.mk]: addressed

**Comment by marian-pritsak** [comment]
So basically extra_debs are same as derived debs except they are devoid of dependency.
Looks good to me.

**Pull request merged**